Yoast/wordpress-seo
# Amitsidhpura prs merged

## Context


* This merges both https://github.com/Yoast/wordpress-seo/pull/15104 and https://github.com/Yoast/wordpress-seo/pull/15113 and adds a few minor changes on top to ensure partial titles are not saved to the database.

## Summary

This PR can be summarized in the following changelog entry:

* Adds the `term_id` for terms in the object that can be filtered with the `wpseo_breadcrumb_links` and `wpseo_breadcrumb_single_link_info` filters. Props to [amitsidhpura](https://github.com/amitsidhpura)
* Fixes a bug where no or incorrect breadcrumbs where shown for 404 pages, author archives and date archives. Props to [amitsidhpura](https://github.com/amitsidhpura)
* [non-user-facing] `breadcrumb_title` is now filled for 404 page. Props to [amitsidhpura](https://github.com/amitsidhpura)
* [non-user-facing] `breadcrumb_title` is no longer filled for date archives. Props to [amitsidhpura](https://github.com/amitsidhpura)

## Test instructions

This PR can be tested by following these steps:

* This PR fixes breadcrumbs for:
  * Date archives
  * Author archives
  * 404 pages.
* This PR also changes the breadcrumbs for:
  * Posts
  * Post type archives
  * Terms
  * Search results
* These should all be tested to ensure no regressions are present.

## UI changes

## Documentation

## Quality assurance
